### ISSUE TITLE
Python pandas datetime plot xticks in unexpected location

### ISSUE THREAD

**Issue by hagaishalev** [matplotlib/matplotlib#12925]
### Bug report
I wouldn't have expected to see the 29th of some of the month labelled
**Bug summary**
I wouldn't have expected to see the 29th of some of the month labelled

**Code for reproduction**

```python
# Paste your code here
#
#
import pandas as pd
import matplotlib.pyplot as plt
x = pd.date_range(start='1/1/2018', periods=80) # 80 days
y = pd.np.random.random(80)
plt.plot(x, y ,'o')
plt.xticks(rotation='vertical')
```

**Actual outcome**
![timeplot](https://user-images.githubusercontent.com/23124908/49356219-b3361b00-f6d3-11e8-97d3-d5cb6b8bc6ad.PNG)

```
# If applicable, paste the console output here
#
#
```

**Expected outcome**

**Matplotlib version**

  * Operating system: Win10 64bit
  * Matplotlib version:  3.0.1
  * Matplotlib backend (`print(matplotlib.get_backend())`): module://ipykernel.pylab.backend_inline
  * Python version: 3.7.1
  * Jupyter version (if applicable): 5.7.2
  * Other libraries: pandas 0.23.4
  * conda 4.5.11

**Comment by jklymak** [comment]
Yes agreed this is a strange choice.  Fixed in #10841 which will hopefully be part of 3.1

**Comment by hagaishalev** [comment]
it does on my 
pandas '0.24.1'
matplotlib '3.0.2'
![image](https://user-images.githubusercontent.com/23124908/52641590-df46f700-2ee1-11e9-8d77-d9c6aef4f0b5.png)
